### PR TITLE
Ch project initialization

### PR DESCRIPTION
**Pull request by Aimeana100**
# 4dbd6655a2781f0109e0528aea4307e361962882 Project set up

Setting up the project structure to help start building RESTful web APIs using Express framework with best structure practices
>
#### What does this PR do?

This is the initial setup for the project [ecommerce-app-legends-bn](https://github.com/atlp-rwanda/ecommerce-app-legends-bn), It should be well files structered, 

-  `src:` This folder contains all the source code of the application.
 --    `api:` This folder contains all the API routes.
 --   `config:` This folder contains configuration files for the application.
 --    `controllers:` This folder contains the business logic of the application.
 --    `models:` This folder contains the data models of the application.
 --    `middlewares:` This folder contains all the middleware functions for the application.
 --    `utils:` This folder contains all the utility functions for the application.
- `docs:` This folder contains all the documentation files for the application.
- `node_modules:` This folder contains all the installed dependencies for the application.
- `package.json:` This file contains the metadata of the application and its dependencies.
- `README.md:` This file contains the documentation of the application.

*N.B* : all files may not be all set as it should as others are to be created in processes as app develpment goes on.

#### Description of Task to be completed

-  Project structure is set up [👌](https://emojipedia.org/ok-hand/)
-  Welcome EndPoint is set up [👌](https://emojipedia.org/ok-hand/) 
-  Set Eslint for codes anlysis  [👌](https://emojipedia.org/ok-hand/)
-  Configure API Doc with swagger  [👌](https://emojipedia.org/ok-hand/)

#### How should this be manually tested?

- clone repo , branch `git clone https://github.com/atlp-rwanda/ecommerce-app-legends-bn.git`
- run `npm i`
- configure env variables `.env` with respect to `.env.example` 
- start the app in development mode `npm run dev`

#### Screenshots
Screenshot 2023-02-27 at 21 13 56
image

image

### CODE REVIEW

File: docs/home.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
+export default {
+  // test
```
Comment (mansurissa) [inline docs/home.js]: remove commented codes

**Review by BenoitIT — COMMENTED** [review]
good work

**Comment by FabCode67** [comment]
@Aimeana100  you did a great job but I think you know that you have to put in consideration what we have discussed about `eslint` configurations.

**Review by FabCode67 — COMMENTED** [review]
@Aimeana100 you did a great job but I think you know that you have to put in consideration what we have discussed about eslint configurations.

File: package.json [hunk 2]
```diff
@@ -0,0 +1,57 @@
+{
+  "name": "ecommerce-app-legends-bn",
+  "version": "1.0.0",
+  "description": "## Introduction",
+  "main": "index.js",
+  "scripts": {
+    "build": "babel src/index.js -d dist",
+    "start": "npm run build && nodemon dist/index.js",
+    "dev": "npm run build && nodemon --exec babel-node src/index.js",
+    "lintfix": "eslint src --fix --cache",
```
Comment (mukunzidd) [inline package.json]: make the script read `"lint:fix: ...."`

File: package.json [hunk 3]
```diff
@@ -0,0 +1,57 @@
+{
+  "name": "ecommerce-app-legends-bn",
+  "version": "1.0.0",
+  "description": "## Introduction",
+  "main": "index.js",
+  "scripts": {
+    "build": "babel src/index.js -d dist",
+    "start": "npm run build && nodemon dist/index.js",
```
Comment (mukunzidd) [inline package.json]: - remove the build command in the start script. 
-  also don't use nodemon to run the built app. Just use the good old node command.

File: package.json [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    "dev": "npm run build && nodemon --exec babel-node src/index.js",
```
Comment (mukunzidd) [inline package.json]: you don't need to build the app in development just run you ap with nodemon and babel. So remove the first command in the chain (`npm run build`).

File: src/config/config.json [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+    "dialect": "mysql"
+  },
+  "production": {
+    "username": "root",
+    "password": null,
+    "database": "database_production",
+    "host": "127.0.0.1",
+    "dialect": "mysql"
+  }
+}
```
Comment (mukunzidd) [inline src/config/config.json]: For the person who worked on setting up the database and environments, they need to use the doting to manage these names.

So this does not concern you , and this is why you should bring in unfinished work aka unmerged changes.

File: src/migrations/20230228111330-create-test.js [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+      updatedAt: {
+        allowNull: false,
+        type: Sequelize.DATE
+      }
+    });
+  },
+  async down(queryInterface, Sequelize) {
+    await queryInterface.dropTable('tests');
+  }
+};
```
Comment (mukunzidd) [inline src/migrations/20230228111330-create-test.js]: What does this test table do?

**Review by mukunzidd — DISMISSED** [review]
Take care of all the review comments. I need you to remove all the files that should not be included in the project setup.

File: src/config/config.json (hunk 5 above)
Comment (Aimeana100) [reply src/config/config.json]: > For the person who worked on setting up the database and environments, they need to use the doting to manage these names.
> 
> So this does not concern you , and this is why you should bring in unfinished work aka unmerged changes.

Thank you @mukunzidd, @mansurissa  for your review and comments, I would like notify you that I have updated and resolved all the issues that has been raised before.  

Kindly you can re-review and merge my ``pull request`` to the `develop` if no other problem. Thanks.! [🙏](https://emojipedia.org/folded-hands/)

image

**Review by Kevin-ishimwe — CHANGES_REQUESTED** [review]
# rules problems and conflicts

the work is decent but the esLint rules are not followed in almost all the files
so think about changing the rules or the code i have also noticed some conflicts when i pulled your work 

rules

the conflicts happen when i pull from this branch 

Capture

**Comment by Aimeana100** [comment]
> # rules problems and conflicts
> the work is decent but the esLint rules are not followed in almost all the files so think about changing the rules or the code i have also noticed some conflicts when i pulled your work
> 
> rules
> 
> the conflicts happen when i pull from this branch
> 
> Capture

Sorry guys, I would like to suggest  solutions in conversation. if you did found some. 
>This is becouse of I dont have eslint issue in my code make sure your works the issues is not being coused by your works.

**Comment by Aimeana100** [comment]
> > # rules problems and conflicts
> > the work is decent but the esLint rules are not followed in almost all the files so think about changing the rules or the code i have also noticed some conflicts when i pulled your work
> > rules
> > the conflicts happen when i pull from this branch
> > Capture
> 
> Sorry guys, I would like to suggest solutions in conversation. if you did found some.
> 
> > This is becouse of I dont have eslint issue in my code make sure your works the issues is not being coused by your works.

@Kevin-ishimwe Due to confussion that we all had, and mistakes of merging the codes from wrong branches. i would like you to clone the repo for testing and reviewing!! 

### [❌](https://emojipedia.org/cross-mark/) Dont pull as you even had already did modification on your side.

**Comment by Aimeana100** [comment]
@Kevin-ishimwe,  @Evaristekanova  , @BenoitIT,   @mukunzidd ,  @mansurissa  @FabCode67 @chrill-christella  Please re-review my work.

*I have updated my works and Eslint & prenttier are pretty running on both `macOs` , `windows`*

**Review by mansurissa — APPROVED** [review]
LGTM

**Review by BenoitIT — COMMENTED** [review]
you would have added package-lock.json in gitignore

**Comment by FabCode67** [comment]
> 

Why can he put  `package-lock.json` in gitignore @BenoitIT ?, I think it will be used in git actions...

**Review by FabCode67 — APPROVED** [review]
Great job @Aimeana100 , In my Laptop  It runs smoothly 🤞

**Review by Kevin-ishimwe — APPROVED** [review]
seems alright at this point

**Review by mansurissa — APPROVED** [review]

**Comment by mansurissa** [comment]
> good work

**Pull request merged**